test/e2e/apimachinery/custom_resource_definition.go

```diff
@@ -42,29 +51,153 @@ var _ = SIGDescribe("CustomResourceDefinition resources", func() {
 				}
+			}
+
+			// Use delete collection to remove the CRDs
+			err = fixtures.DeleteCustomResourceDefinitions(selectorListOpts, apiExtensionClient)
+			framework.ExpectNoError(err, "deleting CustomResourceDefinitions")
+		})
+
+		/*
+			Release : v1.16
+			Testname: Custom Resource Definition, status sub-resource
+			Description: Create a API extension client, create a custom resource definition and then read, update and patch its status sub-resource. API server MUST be able to perform the operations against the status sub-resource.
+		*/
+		ginkgo.It("getting/updating/patching custom resource definition status sub-resource works ", func() {
+			config, err := framework.LoadConfig()
+			framework.ExpectNoError(err, "loading config")
+			apiExtensionClient, err := clientset.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing apiExtensionClient")
+			dynamicClient, err := dynamic.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing dynamic client")
+
+			// Create CRD and waits for the resource to be recognized and available.
+			crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+			crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+			framework.ExpectNoError(err, "creating CustomResourceDefinition")
+			defer func() {
+				err = fixtures.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
 			}()
+
+			// Use dynamic client to test status sub-resource since typed client does not expose status resource.
+			// TODO: Range over both "v1beta" and "v1" once v1 is enabled
+			gvr := schema.GroupVersionResource{Group: v1beta1.GroupName, Version: "v1beta1", Resource: "customresourcedefinitions"}
+			resourceClient := dynamicClient.Resource(gvr)
+			u, err := resourceClient.Get(crd.GetName(), metav1.GetOptions{}, "status")
+			framework.ExpectNoError(err, "getting CustomResourceDefinition status")
+			status := unstructuredToCrd(u)
+			if !equality.Semantic.DeepEqual(status.Spec, crd.Spec) {
+				e2elog.Failf("Expected CustomResourceDefinition Spec to match status subresource Spec, but got:\n%s", diff.ObjectReflectDiff(status.Spec, crd.Spec))
+			}
+
+			updateCondition := v1beta1.CustomResourceDefinitionCondition{Message: "updated"}
+			status.Status.Conditions = append(status.Status.Conditions, updateCondition)
+			u, err = resourceClient.Update(crdToUnstructured(status), metav1.UpdateOptions{}, "status")
+			framework.ExpectNoError(err, "updating CustomResourceDefinition status")
+			updated := unstructuredToCrd(u)
+			expectCondition(updated.Status.Conditions, updateCondition)
+
+			patchCondition := v1beta1.CustomResourceDefinitionCondition{Message: "patched"}
+			u, err = resourceClient.Patch(
+				crd.GetName(),
+				types.JSONPatchType,
+				[]byte(`[{"op": "replace", "path": "/status/conditions", "value": [{"message": "patched"}]}]`),
```

Comment:
won't this stomp system-added conditions? that's probably not something we want to do/demonstrate

Reply:
Good point. I'll tune this to not stomp.